### PR TITLE
add qhull and update matplotlib

### PR DESCRIPTION
**Pull request by wolfv**
Adding qhull seemed pretty easy. 
Matplotlib - unfortunately - does not seem to work yet in Jupyterlite.

The error I am getting is:

```
LinkError: WebAssembly.instantiate(): Import #190 "env" "time": imported function does not match the expected type
```

### CODE REVIEW

**Comment by DerThorsten** [comment]
often a missing `-sWASM_BIGINT` give erros like "imported function does not match the expected type" @wolfv

**Comment by ianthomas23** [comment]
From version 3.9.0 Matplotlib switched to using `meson`, and I think more changes are required here for this. For example, there is no `setupext.py` any more, and it probably needs a meson cross-compilation file (as other `meson`-using packages do here). I have a WIP branch https://github.com/emscripten-forge/recipes/tree/matplotlib_3.9.0 that builds but fails the tests (probably because the build is wrong), and you are welcome to take any of that to use here.

**Comment by wolfv** [comment]
Thanks @DerThorsten - with WASM_BIGINT it seems to work like a charm (for the two things I tested at least). Screenshot attached.

Screenshot 2024-07-15 at 18 49 36

**Comment by wolfv** [comment]
3D triangulation stuff also seems to work?! 

Screenshot 2024-07-15 at 18 53 23

**Comment by DerThorsten** [comment]
> 3D triangulation stuff also seems to work?!
> 
> Screenshot 2024-07-15 at 18 53 23

nice! 
Is this ready for review/merging?

**Comment by wolfv** [comment]
I think so! :)

**Comment by wolfv** [comment]
Although I think it can be simplified some more ...

**Comment by wolfv** [comment]
Ok, so I spent a little time on simplifications but it didn't completely pan out. 

I tried to remove `numpy` and `pybind11` from the `build` dependencies. 

Instead we configure the location in the meson cross file. 

But _note_: the cross-python package does some magic upon activation (it moves all headers and shared objects in the `$BUILD_PREFIX`). So the "final" location of the numpy include dir is _still_ in the $BUILD_PREFIX).

File: recipes/recipes_emscripten/matplotlib/emscripten.meson.cross [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# binaries section is at the end to be able to append the python binary.
```
Comment (ianthomas23) [inline recipes/recipes_emscripten/matplotlib/emscripten.meson.cross:1]: This and the next line could be removed as it is no longer true.

Comment (wolfv) [reply recipes/recipes_emscripten/matplotlib/emscripten.meson.cross]: removed!

**Review by DerThorsten — APPROVED** [review]

**Review by ianthomas23 — APPROVED** [review]

**Pull request merged**